### PR TITLE
numpy.linspace broken with Quantities

### PR DESCRIPTION
**Pull request by mhvk**
Numpy's linspace used to work nicely with quantities:

**Numpy 1.9.3**

```python
>>> np.linspace(-1 * u.m, 1 * u.m, 10)
<Quantity [-1.        ,-0.77777778,-0.55555556,-0.33333333,-0.11111111,
            0.11111111, 0.33333333, 0.55555556, 0.77777778, 1.        ] m>
```

**Numpy 1.10**

```
>>> np.linspace(-1 * u.m, 1 * u.m, 10)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/tom/miniconda3/envs/np110/lib/python2.7/site-packages/numpy/core/function_base.py", line 115, in linspace
    y[-1] = stop
ValueError: setting an array element with a sequence.
```

@mhvk - is this a bug in Astropy, or a bug introduced in Numpy?

A lot of user code might rely on this, so we should get a workaround out soon unless it has to be fixed in Numpy.

### CODE REVIEW

**Comment by mhvk** [comment]
@eteq - I went ahead and made this into a PR with a small documentation update. Only thing I'm not quite sure about is whether I set the labels right...

File: docs/known_issues.rst [hunk 1]
```diff
@@ -99,6 +99,13 @@ An easy solution is::
     array([False], dtype=bool)
 
 
+Quantities in np.linspace failure on numpy 1.10
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In numpy 1.10, `~numpy.linspace` initially failed to work with quantities.
```
Comment (astrofrog) [inline docs/known_issues.rst]: @mhvk - I would rephrase to say 'linspace does not work correctly with quantities when using Numpy 1.10.0 to 1.10.5 due to a bug in Numpy. The solution is to upgrade to Numpy 1.10.6, in which the bug was fixed'

**Comment by mhvk** [comment]
@astrofrog - agreed, might as well be clear about where the problem was. Now changed the text.

**Comment by mhvk** [comment]
I'm going ahead and merge this small documentation update -- there's a bit of a cost to leaving things like this open, if only that they keep popping up any time I look at my open issues/PRs.

**Comment by eteq** [comment]
Backported to v1.0.x in 542f8c3338fd74eacea4f816675756bcb66ce93e and v1.1.x in 12cbe2f3a41253085ea317109fe1d5c6d44dca5b
